### PR TITLE
[WIP] Public Cloud: New test for XFS file systems

### PR DESCRIPTION
**Pull request by jlausuch**
- Related ticket: https://progress.opensuse.org/issues/60179

### CODE REVIEW

**Comment by jlausuch** [comment]
First run without blacklist, just executing everything: http://fromm.arch.suse.de/tests/337

File: tests/publiccloud/xfstests.pm [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    my $instance = $self->{my_instance} = $provider->create_instance(use_extra_disk => {size => $disk_size});
+    $instance->wait_for_guestregister();
+
+    $instance->scp('/tmp/partitions.sh', 'remote:' . '/tmp/partitions.sh');
+    $instance->scp('/tmp/run.sh',        'remote:' . '/tmp/run.sh');
+    $instance->run_ssh_command(cmd => "sudo chmod +x  /tmp/partitions.sh");
+    $instance->run_ssh_command(cmd => "sudo chmod +x  /tmp/run.sh");
+
+    $instance->run_ssh_command(cmd => 'sudo SUSEConnect -r ' . $REG_CODE,         timeout => 600) if (get_required_var('FLAVOR') !~ m/On-Demand/);
+    $instance->run_ssh_command(cmd => 'sudo SUSEConnect -p sle-sdk/12.5/x86_64 ', timeout => 300);
```
Comment (SeroSun) [inline tests/publiccloud/xfstests.pm]: Hard code 12.5 and x86_64 better to change to use FAVOR(adjust needed) and ARCH.

File: data/publiccloud/xfstests/partitions.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -e
+
+disk=$1
+
+parted -s /dev/${disk} mklabel gpt
+parted -s /dev/${disk} mkpart primary 0% 50%
+parted -s /dev/${disk} mkpart primary 50% 100%
+
+mkfs.xfs /dev/${disk}1
```
Comment (SeroSun) [inline data/publiccloud/xfstests/partitions.sh]: /dev/${disk}1 and 2 depend on init status for the disk. I'm not sure this gonna work in all disk or not. Better to check before use.

File: tests/publiccloud/xfstests.pm (hunk 1 above)
Comment (jlausuch) [reply tests/publiccloud/xfstests.pm]: sure, I was testing on 12.5 and forgot to make it generic.

File: data/publiccloud/xfstests/partitions.sh (hunk 2 above)
Comment (jlausuch) [reply data/publiccloud/xfstests/partitions.sh]: the status will be always new and unpartitioned and unformatted. So, I guess we will have always sdX1 and sdX2

**Pull request closed without merging**